### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by enolfc**
# Summary 

Basic version with token refresh and endpoint list. 

----

**Related issue :**

### CODE REVIEW

**Comment by enolfc** [comment]
Still missing documentation, but basics working.

Also lots of repetition in the code (e.g. dealing with input args, but didn't have the time/energy to refactor)

File: egicli/endpoint.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+
+def find_endpoint(service_type, production=True, monitored=True, site=None):
+    q = {"method": "get_service_endpoint", "service_type": service_type}
+    if monitored:
+        q["monitored"] = "Y"
+    if site:
+        q["sitename"] = site
+    url = "?".join([GOCDB_PUBLICURL, parse.urlencode(q)])
+    # XXX UGLY, but GOC is using a IGTF CA not in standard distros
```
Comment (gwarf) [inline egicli/endpoint.py]: Doest it refer to disabling certificate validation that was removed?

**Comment by gwarf** [comment]
Looks great!
Should/can we merge a first version so that it can already be used/tested easily?

File: egicli/endpoint.py (hunk 1 above)
Comment (enolfc) [reply egicli/endpoint.py]: yes, no longer applies

**Comment by enolfc** [comment]
I'm trying to get this today ready for proper review

**Comment by enolfc** [comment]
@gwarf take a look, if you think things look good, we can merge and start improving from here

**Review by gwarf — APPROVED** [review]
Looks great for an initial commit, perfect!

**Pull request merged**